stores/gcpcm/gcpcm.go

```diff
@@ -164,7 +164,7 @@ func (s *GCPStore) Update(secret *corev1.Secret) error {
 	}
 	s.client = client
 	// if there is no secret name, this is the first time we are sending to GCP, create
-	if s.CertificateName == "" {
```

Comment:
s.CertificateName always has value because secretToGCPCert executes first and set value if empty. This produces an error when certificate doesn't exists because always is trying an update.